### PR TITLE
D3 visualization

### PR DESCRIPTION
**Pull request by biancadanforth**
Initial D3 visualization draws first and third party requests to the page.

### CODE REVIEW

File: js/lightbeam.js [hunk 1]
```diff
@@ -1,9 +1,53 @@
...
+  // set up area of the page to put the graph
+  const svg = d3.select('svg');
+  const width = svg.attr('width');
+  const height = svg.attr('height');
+
+  // set up simulation
+  const simulation = d3.forceSimulation().nodes(nodes_data);
+
+  // add forces
+  simulation
```
Comment (jonathanKingston) [inline js/lightbeam.js]: Why is this not in viz?

File: js/lightbeam.js [hunk 2]
```diff
@@ -1,9 +1,53 @@
...
+      // third parties likely don't have favicons
+      node.favicon = '';
+      node.party = 'third';
+      nodes_data.push(node);
+      links_data.push(link);
+    }
+  }
+
+  // set up area of the page to put the graph
+  const svg = d3.select('svg');
```
Comment (jonathanKingston) [inline js/lightbeam.js]: This should be in viz no?

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Ignore external libraries
+js/lib/*
```
Comment (jonathanKingston) [inline .eslintignore]: Use npm instead?

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
 .DS_Store
 node_modules
 npm-debug.log
+js/lib/*
```
Comment (jonathanKingston) [inline .gitignore]: If moving to npm remove this.

File: js/lightbeam.js [hunk 5]
```diff
@@ -1,9 +1,53 @@
 async function renderGraph() {
-  const canvas = document.getElementById('canvas');
-  const context = canvas.getContext('2d');
   const websites = await store.getAll();
 
-  viz.draw(context, websites);
+  // reformat data for D3
```
Comment (jonathanKingston) [inline js/lightbeam.js]: This looks like it is coupled to the visualisations. It shouldn't be.

This code might:
- Change hostname paths and clean replicated data
- Query first party and third parties in a an order it chooses

It shouldn't be changing output to match how a rendering might display it.

File: js/lightbeam.js [hunk 6]
```diff
@@ -1,9 +1,53 @@
...
+    for (const thirdParty in websites[firstParty].thirdPartyRequests) {
+      const node = {};
+      const link = {};
+      link.source = firstParty;
+      link.target = thirdParty;
+      node.hostname = thirdParty;
+      // third parties likely don't have favicons
+      node.favicon = '';
+      node.party = 'third';
+      nodes_data.push(node);
```
Comment (jonathanKingston) [inline js/lightbeam.js]: I thought we were using camel case?

File: js/lightbeam.js (hunk 5 above)
Comment (biancadanforth) [inline js/lightbeam.js]: What are you proposing I do here instead?

File: js/lightbeam.js (hunk 1 above)
Comment (biancadanforth) [inline js/lightbeam.js]: Moved

File: js/lightbeam.js (hunk 2 above)
Comment (biancadanforth) [inline js/lightbeam.js]: Moved

File: js/lightbeam.js (hunk 6 above)
Comment (biancadanforth) [inline js/lightbeam.js]: Yes, we are. This was just based on a tutorial I was using at the time. Changed!

File: .eslintignore (hunk 3 above)
Comment (biancadanforth) [inline .eslintignore]: If I use npm, how do I include the d3 library script in the viz.js file? All solutions so far use `require` or `import` (would need to transpile).

**Pull request closed without merging**